File: ex1.py
def printAsterisk(number):   
  for i in range (number-1):
      value=(2*i)+1
      space=number-i
      print(" "*space+'*'* value,end='\n''\n')
      
   
  
  for j in range (number-1,-1,-1):
      space=number-j
      value=(2*j)+1
      print(" "*space+'*'* value,end='\n''\n') #string multiplication costs O(n)
def main():
  number = int(input('Enter a number of rows:'))
  printAsterisk(number) 

main()

#o(n^2) n being the number passed to func printAsterisk

File: ex1.js
function printAsterisk(number) {
  for (let i = 0; i < number - 1; i++) {
    value = 2 * i + 1;
    space = number - i;
    console.log(" ".repeat(space) + "*".repeat(value) + "\n");
  }
  for (let j = number - 1; j > 0 - 1; j--) {
    space = number - j;
    value = 2 * j + 1;
    console.log(" ".repeat(space) + "*".repeat(value) + "\n");  //repeat has the time complexity of (n) n is the velue of the var value in this case
  }
}
const readline = require("readline").createInterface({  //code sniplet (function) from codeacademy.com
  input: process.stdin,
  output: process.stdout,
});
readline.question("Enter a number: ", (number) => {
  num = parseInt(number);
  printAsterisk(number);
  readline.close();
});

//O(N^2)N being the number passed to function printAsterisk 
